lib/src/kbasesearchengine/main/IndexerWorker.java

```diff
@@ -841,18 +801,139 @@ private void indexObjectWrapperFn(final List<Object> input)
+         * @return
+         * @throws IndexingException
+         * @throws InterruptedException
+         * @throws RetriableIndexingException
+         */
+        private List<ObjectData> getObjectsByIds(final List<GUID> objectRefPath,
+                                                 final Set<GUID> unresolvedGUIDs)
+                throws IndexingException, InterruptedException, RetriableIndexingException {
+            final List<ObjectData> data = new ArrayList<>();
+
+            for(GUID unresolvedGUID: unresolvedGUIDs) {
+                // resolve reference
+                ResolvedReference resRef =
+                        resolveRefs(objectRefPath, ImmutableSet.of(unresolvedGUID)).
+                                                                     iterator().next();
+
+                // add subObjType and subObjId from the unresolved guid as it is
+                // not contained in the resolved reference
+                GUID guid = resRef.getResolvedReference();
+                guid = new GUID(guid.getStorageCode(),
+                                guid.getAccessGroupId(),
+                                guid.getAccessGroupObjectId(),
+                                guid.getVersion(),
+                                unresolvedGUID.getSubObjectType(),
+                                unresolvedGUID.getSubObjectId());
+                resRef = new ResolvedReference(resRef.getReference(),
+                                               guid,
+                                               resRef.getType(),
+                                               resRef.getTimestamp());
+
+                // get object from data source
+                final File tempFile;
+                try {
+                    FileUtil.getOrCreateSubDir(rootTempDir, guid.getStorageCode());
+                    tempFile = File.createTempFile("ws_srv_response_", ".json");
+                } catch (IOException ex) {
+                    throw new RetriableIndexingException(ErrorType.OTHER, ex.getMessage(), ex);
+                }
+
+                try {
+                    // make a copy to avoid mutating the caller's path
+                    final EventHandler handler = getEventHandler(guid);
+                    final SourceData obj = handler.load(
+                            Arrays.asList(resRef.getResolvedReference()), tempFile.toPath());
+
+                    final ObjectTypeParsingRules rules = getObjectTypeParsingRules(resRef);
+
+                    if(rules == null) {
+                        throw new FatalIndexingException(ErrorType.OTHER,
+                                "Unable to find parsing rules for reference "+resRef);
+                    }
+
+                    final ParseObjectsRet parsedRet = parseObjects(guid, this,
+                            Arrays.asList(resRef.getResolvedReference()), obj, rules);
+
+                    final GUID parsedGUID = parsedRet.guidToObj.keySet().iterator().next();
```

Comment:
IIUC, this is taking the first entry in the results of the parse and only processing that. If I'm right, why are all the other possible entries ignored?

Reply:
I have cleaned up this code. Hopefully, I have it right now.

Reply:
I'm still confused here. ParseObjectRet can contain many entried in guidToObj, but here all but the first one (which is arbitrary since it's a map) are ignored. That seems like a bug unless I'm missing something.

Reply:
Fixed this with an iterator.